src/DGtal/base/Clone.h

```diff
@@ -398,14 +423,43 @@ namespace DGtal
       case COW_PTR:
 	return CowPtr<T>( * static_cast< const CowPtr<T>* >( myPtr ) );
       case COUNTED_PTR:
-	return CowPtr<T>( * static_cast< const CountedPtr<T>* >( myPtr ) );
+	return CowPtr<T>( * static_cast< const CountedPtr<T>* >( myPtr ), true );
       case RIGHT_VALUE_REF:
 	return CowPtr<T>( new T( std::move( * const_cast<T*>( static_cast< const T* >( myPtr ) ) ) ) );
       default: ASSERT( false && "[Clone::operator CowPtr<T>() const] Invalid cast for given type. " );
         return CowPtr<T>( 0 );
       }
     }
 
+    /**
+       Cast operator to a smart pointer on T instance. The object is duplicated or not
+       depending on the type of input parameter.
+
+      - const T & -> CountedPtr<T>     // immediate duplication 
+      - T* -> CountedPtr<T>            // acquired              
```

Comment:
Shouldn't be duplicated in that case?

Reply:
Every time a pointer is passed as parameter, it is considered as dyn. alloc. and handed as parameter in order to be acquired by the object (cf doc). However, I do agree that perhaps conversion to CountedPtr<T> could be forbidden from a Clone<T>. In the doc, I only talk about a pointer T* acquired by a CowPtr<T> when the user use Clone<T>. What do you think ?

Reply:
Ok for the acquirement of the pointer, thanks to the doc.

About the conversion to a `CountedPtr` when the `Clone` is initialized from a pointer, the problem is that if the created `CountedPtr` has a shorter lifetime than the `Clone`, it will lead to an invalid pointer (freed by `CountedPtr`). Is that a valid usage of `Clone` ?

While searching a realistic example to the case above, may we have to pass a `Clone` to a `(Const)Alias`?

Reply:
But maybe `Clone` is supposed to be transferred into another holder before being used?

Reply:
Arg. I dunno. It should work in most instances but I am not sure.
Here I think it cannot work (or risk of ambiguity) since I have no constructor ConstAlias<T> from Clone<T>.

Reply:
Ok. Anyway, It's not the time to discuss on this particular point, so we will check that later ;)